metadata-io/src/main/java/com/linkedin/metadata/graph/LineageRegistry.java

```diff
@@ -52,14 +52,14 @@ public LineageRegistry(EntityRegistry entityRegistry) {
     // A produces B :     A -> downstream (produces, OUTGOING), B -> upstream (produces, INCOMING)
     for (LineageEdge edge : lineageEdges) {
       if (edge.isUpstream()) {
-        upstreamPerEntity.computeIfAbsent(edge.sourceEntity, (k) -> new HashSet<>())
+        upstreamPerEntity.computeIfAbsent(edge.sourceEntity.toLowerCase(), (k) -> new HashSet<>())
```

Comment:
Why is this required?

Reply:
It's because sometimes we use lowercased entity name and sometimes we don't so need to make sure we lower case on read and write to this map, so both are handled correctly. I missed putting toLowerCase on the write which caused issues only on dataJobs (vs datajobs) Added test cases to cover dataJobs